src/js/components/SelectMultiple/SelectionSummary.js

```diff
@@ -53,70 +53,86 @@ const SelectionSummary = ({
-          </Text>
-        </Box>
-        {(options.length &&
-          (!limit ||
-            !(!value || (value?.length === 0 && selectedValuesDisabled())))) >
-          0 &&
-          (!onMore || (onMore && value?.length !== 0)) && (
-            <Button
-              a11yTitle={
-                showSelectAll
-                  ? `Select all ${options.length} options`
-                  : `${value?.length} options selected. Clear all?`
-              }
-              label={showSelectAll ? 'Select All' : 'Clear All'}
-              onClick={(event) => {
-                if (onChange) {
-                  const nextSelected = options.filter((i, index) =>
-                    showSelectAll
-                      ? !isDisabled(index) || isSelected(index)
-                      : isDisabled(index) && isSelected(index),
-                  );
-                  const nextValue = nextSelected.map((i) =>
-                    valueKey && valueKey.reduce ? applyKey(i, valueKey) : i,
-                  );
-                  onChange(event, {
-                    option: options,
-                    value: nextValue,
-                    selected: nextSelected,
-                  });
-                }
-                if (limit && !showSelectAll) setActiveIndex(0);
-              }}
-              onFocus={() => setActiveIndex(-1)}
-              ref={clearRef}
-            />
-          )}
-      </Box>
-    );
-  }
+  const selectedInSearch = useCallback(
+    () =>
+      options?.filter((option) =>
+        arrayIncludes(value, option, valueKey || labelKey),
+      ),
+    [options, value, valueKey, labelKey],
+  );
+
+  const showSelectAll = !!(
+    value?.length === 0 ||
+    selectedValuesDisabled() ||
+    !value ||
+    selectedInSearch().length === 0
+  );
   return (
-    <Box pad="xsmall">
-      <Text size="small">{`${value?.length || '0'} selected`}</Text>
+    <Box
+      pad={
+        showSelectedInline ? { left: 'xsmall', vertical: 'xsmall' } : 'xsmall'
+      }
+      direction="row"
+      justify="between"
+      gap="small"
+      fill="horizontal"
+      flex={showSelectedInline}
+    >
+      <Box pad={{ vertical: 'xsmall' }} alignSelf="center">
+        <Text size="small">
+          {value?.length === 0 ||
+          onMore ||
+          !value ||
+          (search !== '' && search !== undefined)
+            ? `${value?.length || 0} selected`
+            : `${value?.length || 0} selected of ${options.length}`}
+        </Text>
+      </Box>
+      {(options.length &&
+        (!limit ||
+          !(!value || (value?.length === 0 && selectedValuesDisabled())))) >
+        0 &&
+        (!onMore || (onMore && value?.length !== 0)) && (
+          <Button
+            a11yTitle={
+              showSelectAll
+                ? `Select all ${options.length} options`
+                : `${value?.length} options selected. Clear all?`
+            }
+            label={showSelectAll ? 'Select All' : 'Clear All'}
```

Comment:
Matt and I had an offline discussion about this one and agreed it could be handled in a separate issue. Changing to "Select matches" and "Clear matches" is still affecting the width of the drop container. I think the naming is clearer but we may want to look into sizing of the button, how comfortable are we with affecting the drop container width, etc. So I think this work could be handled separately